### PR TITLE
User verification

### PR DESCRIPTION
**Pull request by Adam2221114**

### CODE REVIEW

File: server/controllers/userController.js [hunk 1]
```diff
@@ -1,107 +1,149 @@
...
+            password: tempUser.hashedPassword,
+            phone: tempUser.phone
+        });
         await user.save();
 
+        // Record the account creation event in the AccessLog
         await AccessLog.create({
-          eventType: 'account_created',
-          userId: user._id,
+            eventType: 'account_created',
```
Comment (OfficialTomJ) [inline server/controllers/userController.js:80]: Indentation issue

File: server/controllers/userController.js [hunk 2]
```diff
@@ -1,107 +1,149 @@
...
-  // Logic for retrieving user profile
-  try {
-    // Fetch user profile based on user ID stored in req.user.id from auth middleware
-    const user = await User.findById(req.user.id).select('-password');
-    res.json({ user });
-  } catch (error) {
-    console.error('Error fetching user profile:', error);
-    res.status(500).json({ msg: 'Server Error' });
-  }
+    try {
```
Comment (OfficialTomJ) [inline server/controllers/userController.js:97]: Indentation issue

File: server/controllers/userController.js [hunk 3]
```diff
@@ -1,107 +1,149 @@
...
-    const userLogs = await AccessLog.find({ userId });
-
-    res.json({ userLogs });
-  } catch (error) {
-    console.error('Error fetching user logs:', error);
-    res.status(500).json({ error: 'Server error' });
-  }
-};
\ No newline at end of file
+    try {
```
Comment (OfficialTomJ) [inline server/controllers/userController.js:137]: Indention issue

File: server/models/Payment.js
Comment (OfficialTomJ) [inline server/models/Payment.js:1]: Rebase file

**Review by OfficialTomJ — CHANGES_REQUESTED** [review]
Rebase Payment.js and fix indentation issues

**Review by OfficialTomJ — CHANGES_REQUESTED** [review]
When testing the features on the branch, the server crashes as Nodemailer is not installed as a dependency. This is due to you manually uploading files (which I've told you to not do multiple times) and not having the package.json tracked in the commit history.

Please commit to branch properly and track ALL file changes to associated src files and dependency trackers. If you are adding new packages, you will have to track changes in the package.json and package-lock.json files.

Screenshot 2024-05-09 at 3 57 57 pm

**Review by OfficialTomJ — CHANGES_REQUESTED** [review]
package.json and package-lock.json files not included

**Comment by Adam2221114** [comment]
Fixed

**Review by OfficialTomJ — APPROVED** [review]

**Pull request merged**